ordered_product/views.py

```diff
@@ -0,0 +1,141 @@
+from .models import OrderedProduct
+from client.models import Client
+from supplier_product.models import SupplierProduct
+from delivery_location.models import DeliveryLocation
+from product.models import Product
+from supplier.models import Supplier
+from django.shortcuts import redirect
+from django.shortcuts import render
+from django.contrib.auth.decorators import user_passes_test
+from django.core.exceptions import ObjectDoesNotExist
+from django.contrib.auth.models import User
+
+
+def client_check(user):
+    try:
+        return Client.objects.get(client_account=user)
+    except ObjectDoesNotExist:
+        return False
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def change_quantity(request, order_id):
```

Comment:
Are you testing all branches of this function (increase, decrease, no change, exception1, expcetion2)?

Reply:
I fixed it.